src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,143 @@
+
+### Creating items in the viewport
+`*cdkVirtualFor` should be used instead of `*ngFor` to create items in the
+`<cdk-virtual-scroll-viewport>`. `*cdkVirtualFor` supports all of the same features as `*ngFor`.
+The simplest usage just specifies the list of items:
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+`*cdkVirtualFor` makes the following context variables available to the template:
+* `index` - The index of the item in the data source.
+* `count` - The total number of items in the data source.
+* `first` - Whether this is the first item in the data source.
+* `last` - Whether this is the last item in the data source.
+* `even` - Whether the `index` is even.
+* `odd` - Whether the `index` is odd.
+
+All of these apply to the index of the item in the data source, not the index in the rendered
+portion of the data.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items;
+                       let index = index;
+                       let count = count;
+                       let first = first;
+                       let last = last;
+                       let even = even;
+                       let odd = odd">
+    {{item}} - {{index}} - {{count}} - {{first}} - {{last}} - {{even}} - {{odd}}
+  </div>
+</cdk-virtual-scroll-viewport>
+```
+
+A `trackBy` function can be specified and works the same as the `*ngFor` `trackBy`. The `index`
+passed to the tracking function will be the index in the data source, not the index in the rendered
+portion.
+
+#### View recycling
+In order to improve performance `*cdkVirtualFor` saves the views it creates in a cache when they are
+no longer needed. This way the next time a new view is needed one can be recycled from the cache
+rather than being created from scratch. The size of this cache can be adjusted using the
+`templateCacheSize` input. Caching can be disabled by setting it to `0`.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items; templateCacheSize: 0">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+#### Specifying items with an Observable or DataSource
+`*cdkVirtualFor` is set up to accept data from an `Array`, `Observable`, or `DataSource`. The
+`DataSource` for the virtual scroll is the same one used by the table and tree components. A
+`DataSource` is simply an abstract class that has two methods: `connect` and `disconnect`. The
+`connect` method will be called by the virtual scroll viewport to receive a stream that emits the
+data array that should be rendered. The viewport will call `disconnect` when the viewport is
+destroyed, which may be the right time to clean up any subscriptions that were registered during the
+connect process.
+
+### Scrolling over fixed size items
+If you're scrolling over a list of items that are all the same fixed size, you can use the
```

Comment:
I would rephrase `When all items are the same fixed size, ...`

(similar for autosize section)